learning/tour-of-beam/frontend/lib/models/node.dart

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:json_annotation/json_annotation.dart';
+
+import 'group.dart';
+import 'unit.dart';
+
+part 'node.g.dart';
+
+@JsonSerializable()
+class NodeModel {
```

Comment:
`Node` is not abstract in backend. It is what `modules` consist of and it contains a `type` and a `group` or a `unit`:

```
'nodes': [
          {
            'type': 'unit',
            'unit': {'unitId': 'guide', 'name': 'Tour of Beam Guide'}
          },
]
```

Reply:
Asked @eantyshev in the chat.

Reply:
So this comes from the server indeed because golang does not support inheritance. Then we will first deserialize whatever comes from the server into `NodeServerModel`, `UnitServerModel`, and `GroupServerModel` as it is now.

But in the app we need inheritance. So we should have `NodeModel`, `UnitModel extends NodeModel`, and `GroupModel extends NodeModel`.

When we deserialize the tree in the server format, we should convert it into a tree in the latter format.